ZEP/instructions/zep0000.md

```diff
@@ -0,0 +1,399 @@
+# Zarr Community Feedback Process - Zarr Enhancement Proposal (ZEP)
+
+--- 
+ 
+## ZEP 0 — Purpose and process
+
+Author: Sanket Verma <svsanketverma5@gmail.com>
+
+Status: Active
+
+Type: Process
+
+Created: 2022-14-03
+
+## What is ZEP?
+
+ZEP stands for Zarr Enhancement Proposal. A ZEP is a design document providing
+information to the Zarr community, describing a modification or enhancement of
+the Zarr storage specification, a new feature for its processes or environment.
+The ZEP should provide specific proposed changes to the Zarr specification and
+a narrative rationale for the specification changes.
+
+We intend ZEPs to be the primary mechanism for evolving the spec, collecting
+community input on major issues and documenting the design decision that has
+gone into Zarr. In addition, the ZEP author is responsible for building
+consensus within the community and documenting dissenting opinions.
+
+Because the ZEPs are maintained as text files in a versioned repository, their
+revision history is the historical record of the feature proposal.
+
+## ZEP audience
+
+The typical primary audience for ZEPs is the developers working on Zarr and its
+various implementations, the Zarr steering council as well as the Zarr
+community.
+
+The broader Zarr community may also choose to use the process to document
+expected feature additions and to manage complex design coordination problems
+that require collaboration across multiple projects.
+
+## ZEP types
+
+> Specification ZEP
+
+Specification ZEPs deal with the changes related to [Zarr Specifications].
+The SPEC ZEP (abbreviation of Specification ZEP) introduces changes in the core
+specification of Zarr. The changes related to core specification follow a
+strict and thorough review process and should be adopted by everyone in the
+Zarr community.
+
+- Core protocol ZEP
+
+Describes a ZEP which involves changes in the core specification of Zarr. Core
+protocol ZEPs (commonly known as Core ZEPs) are a part of SPEC ZEPs and apply to
+[Zarr Specifications] and its various implementations under the
+[zarr_implementations] GitHub repo. The core protocol ZEPs should be adopted by
+every implementation of Zarr and, in general, the overall Zarr community. Core
+ZEPs must go through a thorough review process, including involvement,
+discussion, and voting from the author of the proposed ZEP, Zarr Steering
+Council, author(s) of various Zarr implementations, and open-source
+projects/research groups using Zarr and the general Zarr community. The general
+advice is that everyone should be made aware of changes introduced by core
+ZEPs. Core protocol ZEPs require community consensus, and consumers are typically not
+free to ignore them.
+
+Note: Currently, Specification ZEPs only deal with the core storage
```

Comment:
```suggestion
Note: Currently, Specification ZEPs only deal with the core
```

Is the the "Core protocol", the "Core storage specification" or the "Core specification". Let's pick one and stick with it.

Reply:
I'd like to go with "Core storage specification".

Reply:
We will need to cross check whatever is set here with https://zarr-specs.readthedocs.io/. There, "storage" is more about things like file v. zip v. sharding. But I'm definitely for getting this language correct consistently across the involved repos.

Reply:
Let's go with a more general term for now? 'Core specification'?

Reply:
That's fine. But how is the "core specification" distinct from just the "specification"? Where in the specification does it say "this is the core part"?

Reply:
Agreed. Let's just go with "specification". Seems like that is what is proposed above as well ( https://github.com/zarr-developers/governance/pull/16#discussion_r861841411 )